lib/src/main/java/com/auth0/jwt/JWTVerifier.java

```diff
@@ -51,43 +46,54 @@ static Verification init(Algorithm algorithm) throws IllegalArgumentException {
 
     public static class BaseVerification implements Verification {
         private final Algorithm algorithm;
-        private final Map<String, Object> expectedClaims;
+        private final Map<String, BiPredicate<Claim, DecodedJWT>> expectedChecks;
         private long defaultLeeway;
+        private final Map<String, Long> customLeeways;
         private boolean ignoreIssuedAt;
+        private Clock clock;
 
         BaseVerification(Algorithm algorithm) throws IllegalArgumentException {
             if (algorithm == null) {
                 throw new IllegalArgumentException("The Algorithm cannot be null.");
             }
 
             this.algorithm = algorithm;
-            this.expectedClaims = new HashMap<>();
+            this.expectedChecks = new HashMap<>();
```

Comment:
What do you think of using LinkedHashMap to maintain the insertion order of claims? It would make checks run consistently in the order they are declared in the code.

Reply:
Nice catch @panga! I was actually thinking of it, and in future, we can have multiple checks for a single claim by using an ArrayList instead of a Map. but since that is a behaviour change, I will change this one to LinkedHashMap